pkg/testing/runner/utils.go

```diff
@@ -0,0 +1,42 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package runner
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/elastic/elastic-agent/pkg/core/process"
+)
+
+// WorkDir returns the current absolute working directory.
+func WorkDir() (string, error) {
+	wd, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("failed to get work directory: %w", err)
+	}
+	wd, err = filepath.Abs(wd)
+	if err != nil {
+		return "", fmt.Errorf("failed to get absolute path to work directory: %w", err)
+	}
+	return wd, nil
+}
+
+func AttachOut(w io.Writer) process.CmdOption {
+	return func(c *exec.Cmd) error {
+		c.Stdout = w
+		return nil
+	}
+}
+
+func AttachErr(w io.Writer) process.CmdOption {
+	return func(c *exec.Cmd) error {
+		c.Stderr = w
+		return nil
+	}
+}
```

Comment:
What's the advantage in using this instead of [StdErrPipe](https://pkg.go.dev/os/exec#Cmd.StderrPipe) or [StdOutPipe](https://pkg.go.dev/os/exec#Cmd.StdoutPipe) optionally with an [io.Pipe](https://pkg.go.dev/io#Pipe) if we don't like the ReadCloser types ?

Reply:
The command execution is the same command execution that is used for the core pieces of Elastic Agent. I prefer to use those command execution as it provides OS level specifics about Pkill and Windows Job so if the parent process dies the child process also dies.

Reply:
I fail to see how what  you are describing is related to how we attach to stdout and stderr... 
What I meant was that these two functions are a very thin layer around setting up an `exec.Cmd` and in a lot of places we are using them just to dump output and error in the same buffer. So I was wondering: what was wrong with using directly the functions exposed by `exec.Cmd` ?

Reply:
The difference is because we merge the stdout and stderr into a single buffer where the output is combined. `StdErrPipe` and `StdOutPipe` provide unique buffers.